Edit pr template to include new release note generation process (#5903)
concourse/concourse: .github/pull_request_template.md

<!--
Hi there! Thanks for submitting a pull request to Concourse!

The title of your pull request will be used to generate the release notes.

> Please make if a brief sentence that describes the PR in the imperative form.

Reviewer: There's a typo here ("make if"), but this rewording might be better than just fixing the typo:

```suggestion
Please provide a brief sentence that describes the PR, using the [imperative mood].
```

and then add this below the paragraph:

```md
[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative
```

Please refrain from adding prefixes like 'feature:', and don't include a period

> at the end. We will edit the title if needed so don't worry about getting it

Reviewer: Maybe split the 'We will ...' into a separate paragraph below the examples?

<!--
Hi there! Thanks for submitting a pull request to Concourse!

Please ensure your PR title is appropriate because it will be used to generate

> the release note.

Reviewer: May help to suggest what makes it 'appropriate' - that it should summarize the content of the PR?

Also, what are we going to do about naming conventions like `atc: structure:` or `feat(load_var):`?

Author: Ah good point, I can add in another sentence for describing what it means to be appropriate.

I was hoping to get rid of the naming conventions with the `atc:` or `feat(..)`! Maybe I should mention to not include any prefix within the title in the pr template?

Reviewer: @clarafu Yeah, we could mention that. It's a bit redundant given the template already has a part where the user chooses bug fix / feature / etc., plus we already add labels.

Author: How does this sound 

```
Please ensure your PR title is appropriate because it will be used to generate
the release note. The title should try to briefly summarize the contents of the
PR and should be written as a sentence. Please refrain from adding any prefixes
within the title. An example of a PR title is "Added a new fly command
to disable resource versions".
```

is that a good enough example or should it not be in past tense?
